### PR TITLE
Implement annotation replicator.v1.mittwald.de/strip-labels

### LINKED ISSUE #151: Support removing labels upon replication
**Is your feature request related to a problem? Please describe.**
When installing applications with ArgoCD or similar GitOps systems, detection of association to a managed application is done by a label. In our case we use a Kafka operator to create secrets (based on our GitOps settings), but then replicate those secrets. As this newly replicated secret isn't part of the GitOps-based description of the desired state, those secrets are detected as to-be-deleted.

**Describe the solution you'd like**
A flag that allows removing labels by key during replication.

**Describe alternatives you've considered**
Ignoring resources marked as to-be-deleted, not really an option.

**Additional context**
n/a

### PR DESCRIPTION
**Pull request by jkroepke**
Hi again,

this PR add a new feature to mittwald/kubernetes-replicator. 

Motivation of this change:

Operators like https://github.com/strimzi/strimzi-kafka-operator implements an own garbage collection based on specific labels defined on resources. If mittwald replicator replicate secrets to different namespace, the strimzi-kafka-operator will remove the replicated secrets because from operators point of view the secret is a left-over. To mitigate the issue, set the annotation replicator.v1.mittwald.de/strip-labels=true to remove all labels on the replicated resource.

Fixes #151

### CODE REVIEW

**Comment by jkroepke** [comment]
Fixes https://github.com/mittwald/kubernetes-replicator/issues/151

**Comment by jkroepke** [comment]
Hi @hensur , if you find some time for review this, it would be great!

File: replicate/secret/secrets_test.go [hunk 1]
```diff
@@ -692,6 +765,85 @@ func TestSecretReplicator(t *testing.T) {
 		require.Equal(t, []byte("Hello Bar"), updTarget.Data["bar"])
 	})
 
+	t.Run("replication is pushed to other namespaces and strip labels", func(t *testing.T) {
```
Comment (hensur) [inline replicate/secret/secrets_test.go:768]: To what extend do these tests differ? If they do, please adjust the test description accordingly.

**Review by hensur — CHANGES_REQUESTED** [review]

File: replicate/configmap/configmaps.go [hunk 2]
```diff
@@ -164,6 +169,10 @@ func (r *Replicator) ReplicateObjectTo(sourceObj interface{}, target *v1.Namespa
 		resourceCopy.OwnerReferences = source.OwnerReferences
 	}
 
+	stripLabels, ok := source.Annotations[common.StripLabels]
```
[the remark this answers is not in the document]
Comment (jkroepke) [reply replicate/configmap/configmaps.go]: Oops. Sorry for this. commit dbfc1ca65481cf89c35de8829f8e9ffe913330e4 should resolve this issue.

File: replicate/configmap/configmaps.go [hunk 3]
```diff
@@ -77,6 +77,11 @@ func (r *Replicator) ReplicateDataFrom(sourceObj interface{}, targetObj interfac
 		targetCopy.OwnerReferences = nil
 	}
 
+	stripLabels, ok := source.Annotations[common.StripLabels]
```
[the remark this answers is not in the document]
Comment (jkroepke) [reply replicate/configmap/configmaps.go:80]: by reading the source, I agree with you. For this change, I almost replicate my changes from ownerReference. I will remove this all.

[the remark this answers is not in the document]
Comment (jkroepke) [reply replicate/configmap/configmaps.go]: 7e07b4a resolve this.

File: replicate/role/roles.go [hunk 4]
```diff
@@ -79,6 +79,11 @@ func (r *Replicator) ReplicateDataFrom(sourceObj interface{}, targetObj interfac
 		targetCopy.OwnerReferences = nil
 	}
 
+	stripLabels, ok := source.Annotations[common.StripLabels]
```
[the remark this answers is not in the document]
Comment (jkroepke) [reply replicate/role/roles.go]: 19a1d18 resolve this.

File: replicate/rolebinding/rolebindings.go [hunk 5]
```diff
@@ -80,6 +80,11 @@ func (r *Replicator) ReplicateDataFrom(sourceObj interface{}, targetObj interfac
 		targetCopy.OwnerReferences = nil
 	}
 
+	stripLabels, ok := source.Annotations[common.StripLabels]
```
[the remark this answers is not in the document]
Comment (jkroepke) [reply replicate/rolebinding/rolebindings.go]: 19a1d18 resolve this.

File: replicate/secret/secrets.go [hunk 6]
```diff
@@ -76,6 +76,11 @@ func (r *Replicator) ReplicateDataFrom(sourceObj interface{}, targetObj interfac
 
 	targetCopy := target.DeepCopy()
 
+	stripLabels, ok := source.Annotations[common.StripLabels]
```
[the remark this answers is not in the document]
Comment (jkroepke) [reply replicate/secret/secrets.go]: 19a1d18 resolve this.

File: replicate/secret/secrets_test.go (hunk 1 above)
Comment (jkroepke) [reply replicate/secret/secrets_test.go:768]: They don't.

File: replicate/secret/secrets_test.go [hunk 7]
```diff
@@ -613,6 +613,79 @@ func TestSecretReplicator(t *testing.T) {
...
+		secrets2 := client.CoreV1().Secrets(prefix + "test2")
+		updTarget, err := secrets2.Get(context.TODO(), source.Name, metav1.GetOptions{})
+
+		require.NoError(t, err)
+		require.Equal(t, []byte("Hello Foo"), updTarget.Data["foo"])
+		require.False(t, reflect.DeepEqual(sourceLabels, updTarget.Labels))
+
+		require.Equal(t, map[string]string(nil), updTarget.Labels)
+
+		wg, stop = waitForSecrets(client, 1, EventHandlerFuncs{
```
[the remark this answers is not in the document]
Comment (jkroepke) [reply replicate/secret/secrets_test.go]: Remove in 20112090

**Comment by jkroepke** [comment]
Hi @hensur 

thanks for taking a feedback. Sorry for doing a lot of copypasta, sometime I do not have the full understanding of golang. 

I do a commit for each of you comments/suggestions. I hope you can quicker followup the changes that I do, to validate them.  Let me know, if I should do a rebase/squash.

**Review by hensur — APPROVED** [review]

**Comment by hensur** [comment]
Hey @jkroepke,
I just squashed and merged your changes.

Don't worry, sometimes I don't have that understanding either :)

**Pull request merged**